### DISCUSSION TITLE
Please add undo and redo functions in edit mode.

### DISCUSSION THREAD

**Discussion by jememouse** [langgenius/dify#3551] [Suggestion]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Pleas do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing?

请在编辑模式增加撤销与恢复功能。Please add undo and redo functions in edit mode.

### 2. Describe the feature you'd like to see

请在编辑模式增加撤销与恢复功能。Please add undo and redo functions in edit mode.

### 3. How will this feature improve your workflow or experience?

请在编辑模式增加撤销与恢复功能。Please add undo and redo functions in edit mode.

### 5. Can you help us with this feature?

- [X] I am interested in contributing to this feature.

**Comment by crazywoola** [comment]
Can you paste a screenshot as well? A little bit confused regarding the editing part.

**Reply by Semonxue** [reply]
This function is very useful. If I deleted one or more nodes by mistake, I can't make it back. If the prompt is missing, the workflow will not get the right result. Please add undo/redo or add version control in edtior.
